msticpy/sectools/geoip.py

```diff
@@ -278,27 +293,139 @@ class GeoLiteLookup(GeoIpLookup):
+
+        Parameters
+        ----------
+        url : str
+            Web URL location to the Maxmind city Database. (the default is None)
+        db_folder: str, optional
+            Provide absolute path to the folder containing MMDB file (e.g. '/usr/home' or 'C:\maxmind'). 
+            If no path provided, it is set to download msticpy package installation directory.(the default is None)
+        """
+
+        if url is None:
+            url = self._MAXMIND_DOWNLOAD
+
+        if db_folder is None:
+            db_folder = self._PKG_DIR
+
+        _DB_ARCHIVE_PATH = os.path.join(db_folder, self._DB_ARCHIVE)
+        _DB_FILE_PATH = os.path.join(db_folder, self._DB_FILE)
+
+        try:
+            response = requests.get(url, stream=True)
+            response.raise_for_status()
+        except HTTPError as http_err:
+            print(f"HTTP error occurred: {http_err}")
+        except Exception as err:
+            print(f"Other error occurred: {err}")
+        else:
+            print("Downloading GeoLite DB archive from MaxMind....")
+            with open(_DB_ARCHIVE_PATH, "wb") as fd:
+                for chunk in response.iter_content(chunk_size=10000):
+                    fd.write(chunk)
+                print(f"Downloaded archive location :: {_DB_ARCHIVE_PATH}")
+            try:
+                with gzip.open(_DB_ARCHIVE_PATH, "rb") as f_in:
+                    print(f"Extracting city database...")
+                    with open(_DB_FILE_PATH, "wb") as f_out:
+                        shutil.copyfileobj(f_in, f_out)
+                        print(
+                            f"Extraction complete. Local Maxmind city DB :: {_DB_FILE_PATH}"
+                        )
+            except IOError as err:
+                print(f"{_DB_ARCHIVE_PATH} {err}")
+
+    def get_geolite_dbpath(
+        self, db_folder: str = None, force_update: bool = False
+    ) -> str:
+        """ get the correct path containing GeoLite City Database
+
+        Parameters
+        ----------
+        db_folder: str, optional
+            Provide absolute path to the folder containing MMDB file (e.g. '/usr/home' or 'C:\maxmind').  
+            If no path provided, it is set to download msticpy package installation directory.
+        force_update : bool, optional
+            Force update can be set to true or false. depending on it, new download request will be initiated.
+
+        Returns
+        -------
+        str
+            Returns the absoulte path of local maxmind geolite city database after control flow logic.
+
+        """
+        if db_folder is None:
+            db_folder = self._PKG_DIR
+        force_update = self._force_update
+        _list_of_db_paths = glob.glob(db_folder + "/*.mmdb")
+
+        if _list_of_db_paths:
+            _latest_db_path = max(_list_of_db_paths, key=os.path.getmtime)
+            print(f"Local Maxmind City Database already present in {db_folder}")
```

Comment:
Remove this print() unless you want a verbose option

Reply:
removed